app/views/bookings/show.html.erb

```diff
@@ -15,7 +15,9 @@
       <p>To <%= @booking.end_time%></p>
       <p><%= @booking.status %> by the organiser</p>
       <br>
-      <button type="button" class="btn btn-success"><%= link_to "Edit", edit_activity_path(@activity) %></button>
+      <% if policy(@booking).edit? %>
+        <button type="button" class="btn btn-success"><%= link_to "Edit", edit_activity_booking_path(@activity) %></button>
```

Comment:
@chloechemin Not sure if you resolved this already, but I fixed the edit booking path.